src/main/java/com/example/demo/controller/RecipeController.java

```diff
@@ -0,0 +1,126 @@
+        } catch (UnauthorizedAccessException e) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("You are not authorized to delete this recipe. Only Admin or Creator can");
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("An error occurred while deleting the recipe");
+        }
+    }
+
+    @GetMapping("byName/{recipeName}")
+    public RecipeDtoRegular getRecipeByName(@PathVariable String recipeName){
+        return recipeService.searchRecipeByName(recipeName);
+    }
+
+
+    @GetMapping("byUser/{userId}")
+    public List<RecipeDtoRegular> getRecipesByUser(@PathVariable int userId){
+        return recipeService.getRecipeDtosByUser(userId);
+    }
+
+    @GetMapping("publicRecipes")
+    public List<RecipeDtoRegular> getAllPublicRecipes(){
+        return recipeService.getAllPublicRecipes(true);
+    }
+
+    @GetMapping("/ingredient/{ingredient}")
+    public List<RecipeDtoRegular> getRecipesByIngredient(@PathVariable String ingredient){
+        return recipeService.searchRecipesByIngredients(ingredient);
+    }
+
+    @PostMapping("/protein")
+    public List<RecipeDtoRegular> getRecipesByProteinAndUser(@RequestBody ProteinRequest proteinRequest){
+        return recipeService.searchRecipesByProteinAndUser(proteinRequest.getUserId(), proteinRequest.getMinProtein());
+    }
+
+    @PostMapping("/fat")
+    public List<RecipeDtoRegular> getRecipesByFatAndUser(@RequestBody FatRequest fatRequest){
+        return recipeService.searchRecipesByFatAndUser(fatRequest.getUserId(), fatRequest.getMaxFat());
+    }
+
+    @PostMapping("/calories")
+    public List<RecipeDtoRegular> getRecipesByCaloriesAndUser(@RequestBody CaloriesRequest caloriesRequest){
+        return recipeService.searchRecipesByCaloriesAndUser(caloriesRequest.getUserId(),
+                caloriesRequest.getMinCalories());
+    }
+
+    @PostMapping("/carbohydrates")
+    public List<RecipeDtoRegular> getRecipesByCarbohydratesAndUser(@RequestBody CarbohydratesRequest carbohydratesRequest){
+        return recipeService.searchRecipesByCarbohydratesAndUser(carbohydratesRequest.getUserId(),
+                carbohydratesRequest.getMinCarbohydrates());
+    }
+
+    // For total results
+    @PostMapping("/carbohydratesRange")
+    public List<RecipeDtoRegular> getRecipesByCarbohydratesRangeAndUser(@RequestBody SearchRequest request){
+        return recipeService.searchRecipesByCarbohydratesRangeAndUser(request.getUserId(),
+                request.getMinCarbohydrates(), request.getMaxCarbohydrates());
+    }
+
+    @PostMapping("/proteinsRange")
+    public List<RecipeDtoRegular> getRecipesByProteinsRangeAndUser(@RequestBody SearchRequest request){
+        return recipeService.searchRecipesByProteinsRangeAndUser(request.getUserId(),
+                request.getMinProtein(), request.getMaxProtein());
+    }
+
+    @PostMapping("/fatRange")

```

Comment:
fat-range, please split all multiple word names with `-` in endpoints. This applies to all